RecoJets/JetProducers/plugins/FixedGridRhoProducerFastjetFromRecHit.cc

```diff
@@ -0,0 +1,180 @@
+
+  fastjet::GridMedianBackgroundEstimator bge_;
+  edm::EDGetTokenT<HBHERecHitCollection> hbheRecHitsTag_;
+  edm::EDGetTokenT<EcalRecHitCollection> ebRecHitsTag_;
+  edm::EDGetTokenT<EcalRecHitCollection> eeRecHitsTag_;
+
+  const EgammaHcalIsolation::arrayHB eThresHB_;
+  const EgammaHcalIsolation::arrayHE eThresHE_;
+
+  //Muon HLT currently use ECAL-only rho for ECAL isolation,
+  //and HCAL-only rho for HCAL isolation. So, this skipping option is needed.
+  bool skipHCAL_;
+  bool skipECAL_;
+
+  const edm::ESGetToken<EcalPFRecHitThresholds, EcalPFRecHitThresholdsRcd> ecalPFRechitThresholdsToken_;
+  const edm::ESGetToken<CaloGeometry, CaloGeometryRecord> caloGeometryToken_;
+};
+
+FixedGridRhoProducerFastjetFromRecHit::FixedGridRhoProducerFastjetFromRecHit(const edm::ParameterSet &iConfig)
+    : bge_(iConfig.getParameter<double>("maxRapidity"), iConfig.getParameter<double>("gridSpacing")),
+      hbheRecHitsTag_(consumes(iConfig.getParameter<edm::InputTag>("hbheRecHitsTag"))),
+      ebRecHitsTag_(consumes(iConfig.getParameter<edm::InputTag>("ebRecHitsTag"))),
+      eeRecHitsTag_(consumes(iConfig.getParameter<edm::InputTag>("eeRecHitsTag"))),
+      eThresHB_(iConfig.getParameter<EgammaHcalIsolation::arrayHB>("eThresHB")),
+      eThresHE_(iConfig.getParameter<EgammaHcalIsolation::arrayHE>("eThresHE")),
+      skipHCAL_(iConfig.getParameter<bool>("skipHCAL")),
+      skipECAL_(iConfig.getParameter<bool>("skipECAL")),
+      ecalPFRechitThresholdsToken_{esConsumes()},
+      caloGeometryToken_{esConsumes()} {
+  produces<double>();
+}
+
+void FixedGridRhoProducerFastjetFromRecHit::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+  //We use raw recHits and not the PF recHits, because in Phase1 muon and egamma HLT,
+  //PF recHit collection is regional, while the full raw recHit collection is available.
+  desc.add<edm::InputTag>("hbheRecHitsTag", edm::InputTag("hltHbhereco"));
+  desc.add<edm::InputTag>("ebRecHitsTag", edm::InputTag("hltEcalRecHit", "EcalRecHitsEB"));
+  desc.add<edm::InputTag>("eeRecHitsTag", edm::InputTag("hltEcalRecHit", "EcalRecHitsEE"));
+  desc.add<bool>("skipHCAL", false);
+  desc.add<bool>("skipECAL", false);
+  //eThresHB/HE are from RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitHBHE_cfi.py
+  desc.add<std::vector<double> >("eThresHB", {0.1, 0.2, 0.3, 0.3});
+  desc.add<std::vector<double> >("eThresHE", {0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2});
+  desc.add<double>("maxRapidity", 2.5);
+  desc.add<double>("gridSpacing", 0.55);
+  descriptions.add("hltFixedGridRhoProducerFastjetFromRecHit", desc);
+}
+
+FixedGridRhoProducerFastjetFromRecHit::~FixedGridRhoProducerFastjetFromRecHit() {}
+
+void FixedGridRhoProducerFastjetFromRecHit::produce(edm::Event &iEvent, const edm::EventSetup &iSetup) {
+  std::vector<fastjet::PseudoJet> inputs;
+  auto const &thresholds = iSetup.getData(ecalPFRechitThresholdsToken_);
```

Comment:
```suggestion
  auto const &thresholds = iSetup.getData(ecalPFRecHitThresholdsToken_);
```
Consistent naming (RecHit).